src/tty_util.rs

```diff
@@ -41,13 +41,17 @@ impl TtyGuard {
     pub fn check(&mut self) -> Result<(), io::Error> {
         let &mut TtyGuard { ref tty, my_pgrp, .. } = self;
         tty.as_ref().map_or(Ok(()), |f| {
-            if try!(unsafe { get_group(f.as_raw_fd()) }) == 0 {
-                try!(unsafe { give_tty_to(f.as_raw_fd(), my_pgrp) });
+            let tty_owner_grp = try!(unsafe { get_group(f.as_raw_fd()) });
+            if tty_owner_grp != 0 {
+                let kill_res = unsafe { kill(tty_owner_grp, 0) };
+                if kill_res < 0 && io::Error::last_os_error().raw_os_error().unwrap() == ESRCH {
```

Comment:
There is `errno()` function in libc and nix. Wrapping and unwrapping looks weird.

BTW why don't you use `kill` from `nix` ? It returns an error that you can match against.